Modules/_sqlite/blob.c

```diff
@@ -102,12 +97,18 @@ pysqlite_close_all_blobs(pysqlite_Connection *self)
 {
     for (int i = 0; i < PyList_GET_SIZE(self->blobs); i++) {
         PyObject *weakref = PyList_GET_ITEM(self->blobs, i);
-        PyObject *blob = _PyWeakref_GET_REF(weakref);
-        if (blob == NULL) {
+        if (!PyWeakref_Check(weakref)) {
             continue;
         }
-        close_blob((pysqlite_Blob *)blob);
-        Py_DECREF(blob);
+        PyObject *blob;
+        if (PyWeakref_GetRef(weakref, &blob) < 0) {
+            PyErr_WriteUnraisable((PyObject *)self);
```

Comment:
We should make _PyErr_WriteUnraisableMsg() public, since this one logs a message which is hard to get, it gives no context.